cpp/spherical_volume_rendering_util.cpp

```diff
@@ -0,0 +1,185 @@
+#include "spherical_volume_rendering_util.h"
+#include <vector>
+#include <cmath>
+
+// Takes the absolute value of 'value' and determines if it is less than the tolerance.
+// In this case, we assume tolerance is an arbitrary small value.
+// This is used when floating point mathematics carries rounding error.
+[[nodiscard]] inline bool isNearZero(double value, double tolerance) {
+    return std::abs(value) < tolerance;
+}
+
+// Produces the corresponding angular or azimuthal voxel ID given two directions.
+// TODO(cgyurgyik): Update documentation, provide link to better description.
+[[nodiscard]] inline size_t getInitialVoxelID(double y, double x, size_t num_sections) {
+    const size_t current_voxel_ID = std::floor(std::atan2(y, x) * num_sections / ( 2 * M_PI));
+    return (current_voxel_ID + num_sections) % num_sections;
+}
+
+struct RadialHitParameters {
+    double tMaxR;
+    size_t tStepR;
+    bool previous_transition_flag;
```

Comment:
Is this inert, or does it do something?

Reply:
It will be the return of radial_hit() once I've implemented it.